### ISSUE TITLE
Equal height items

### ISSUE THREAD

**Issue by studioraygun** [ant-design/ant-design#9154]
### What problem does this feature solve?
When using the Card component it would be great to add support for equal heights. Similarly, when using Row and Col, if we could add support for equal heights on all children of a Row item, it would solve a lot of layout issues.

### What does the proposed API look like?
`<Row equal-heights>` or similar

**Comment by yutingzhao1991** [comment]
I don't understand，What does `a lot of layout issues` mean?

Row height === max(Col height) so it will not bring layout issues. https://codesandbox.io/s/my8p6vk7wj

**Comment by studioraygun** [comment]
I mean add a feature to make the columns (and other elements) equal heights inside a row.

Your example currently looks like this:

<img width="538" alt="screen shot 2018-01-29 at 09 33 56" src="https://user-images.githubusercontent.com/7352489/35503321-b2978362-04d7-11e8-8843-ec459c10d2d2.png">

But it would be nice to add `<Row equal-heights>` and the content appear as so:

<img width="535" alt="screen shot 2018-01-29 at 09 35 56" src="https://user-images.githubusercontent.com/7352489/35503387-e55b152a-04d7-11e8-951d-b6f5156d73bc.png">

**Comment by yutingzhao1991** [comment]
I see ... It looks reasonable.

Maybe call `vertical-fill` better ?

**Comment by afc163** [comment]
I don't think we need this, the height of children should leave to children's content or it's style.

**Comment by birendra-90** [comment]
I am getting the same issues. Any have a solution for that?

**Comment by sgwcollins** [comment]
Any resolution on this issue?

**Comment by alvin** [comment]
Using flexbox-enabled rows with type="flex" and setting the height to 100% on the column contents worked well for me.

`<Row gutter={16} type="flex">`

https://ant.design/components/grid/#Row

<img width="918" alt="image" src="https://user-images.githubusercontent.com/16319/47612847-23e87880-da40-11e8-88b7-82599919206c.png">

**Comment by youlinaXealth** [comment]
It doesn't work for me just adding 'flex' to row and add 100% height to column contents.

**Comment by youlinaXealth** [comment]
Thanks Alvin. It works for me to set type='flux'. But don't forget to set the height to 100% on the column contents also. Not column itself, it should be the component under column.

**Comment by rsrogers4** [comment]
I have Cards in my grid. Setting row type to "flux" and the card's height to "100%" works in the sense that each card fills the full height of the row. However, the card's action bar is no longer at the bottom of the row. It looks like this:

![image](https://user-images.githubusercontent.com/508163/57247263-5a193d00-700d-11e9-8590-76f8113fba70.png)

How can I have cards in a grid that expand to the height of the row, and with the card action buttons at the bottom of the row?

**Comment by dianaabv16** [comment]
in the Meta tag you can add a class and specify your fixed height. Like
<Card>
<Meta  
className="cardBodyStyle"
 title={key.title}
description={key.description}
 />
</Card>
css: 
.cardBodyStyle {
    height: 100px;
}

**Comment by milad1367** [comment]
> Using flexbox-enabled rows with type="flex" and setting the height to 100% on the column contents worked well for me.
> 
> `<Row gutter={16} type="flex">`
> 
> https://ant.design/components/grid/#Row
> 
> <img alt="image" width="918" src="https://user-images.githubusercontent.com/16319/47612847-23e87880-da40-11e8-88b7-82599919206c.png">

this is the right way

**Comment by pavloneiman** [comment]
Try this approach to stretch child node to all available space (vertically and horizontally)
```
<Row gutter={16}>
    <Col span={12} style={{display: 'flex'}}>
        <Card style={{flex:1}}>
            1<br/>2<br/>3
        </Card>
    </Col>
    <Col span={12} style={{display: 'flex'}}>
        <Card style={{flex:1}}>
            1<br/>2
        </Card>
    </Col>
</Row>
```

**Comment by cromat** [comment]
For me, none of these were working. The solution was to remove `height: 100%` and set `display: inline-flex` and `align-self: stretch` on Col tags. If you have multiple elements inside Col, you'll need to wrap them within another container so that the Col has only one child.

**Comment by yarinsa** [comment]
> If someone is interested in the case you are using a Card component inside a Col as @studioraygun intended and you have the same issue as @rsrogers4, here is how you get equal heights body cards:
> 
> TLDR :
> 
> * Set `display: flex` on Col and inner Card components
> * Add `flex-direction: column` on inner Card component
> * Set `flex-grow: 1` on `.ant-card-body`
> 
> Example :
> 
> ```
> <Row gutter={16}>
> 
>     <Col style={{ display: 'flex' }}>
>         <Card
>             className='flexible-card'
>             style={{ display: 'flex', flexDirection: 'column' }}
>             actions={[
>                 <SettingOutlined key="setting" />,
>                 <EditOutlined key="edit" />,
>                 <EllipsisOutlined key="ellipsis" />,
>             ]}
>         >
>             <Meta title="Card title" description="This is the description"/>
>         </Card>
>     </Col>
>     
>     <Col style={{ display: 'flex' }}>
>         <Card
>             className='flexible-card'
>             style={{ display: 'flex', flexDirection: 'column' }}
>             actions={[
>                 <SettingOutlined key="setting" />,
>                 <EditOutlined key="edit" />,
>                 <EllipsisOutlined key="ellipsis" />,
>             ]}
>         >
>              <Meta title="Card title" description="This is an even longer description to show the card body growing"/>
>         </Card>
>     </Col>
>     
> </Row>
> ```
> 
> And have written in your css :
> 
> ```
> .flexible-card .ant-card-body {
>     flex-grow: 1
> }
> ```
> 
> Instead of this: ![image](https://user-images.githubusercontent.com/3034969/159312990-1965844f-7df1-4846-b30d-181ff570cc68.png)
> 
> You will get this: ![image](https://user-images.githubusercontent.com/3034969/159313817-bbc23ab3-f574-4853-8083-b898efe17da8.png)

You helped me a lot thanks!!

**Comment by brhanegiday** [comment]
> For me, none of these were working. The solution was to remove `height: 100%` and set `display: inline-flex` and `align-self: stretch` on Col tags. If you have multiple elements inside Col, you'll need to wrap them within another container so that the Col has only one child.

This has worked for me. Also make sure that the child of `Col` have `width:100% `so that it can span fully. 
Thank you

**Comment by sajib-bme-kuet** [comment]
> If someone is interested in the case you are using a Card component inside a Col as @studioraygun intended and you have the same issue as @rsrogers4, here is how you get equal heights body cards:
> 
> TLDR :
> 
> * Set `display: flex` on Col and inner Card components
> * Add `flex-direction: column` on inner Card component
> * Set `flex-grow: 1` on `.ant-card-body`
> 
> Example :
> 
> ```
> <Row gutter={16}>
> 
>     <Col style={{ display: 'flex' }}>
>         <Card
>             className='flexible-card'
>             style={{ display: 'flex', flexDirection: 'column' }}
>             actions={[
>                 <SettingOutlined key="setting" />,
>                 <EditOutlined key="edit" />,
>                 <EllipsisOutlined key="ellipsis" />,
>             ]}
>         >
>             <Meta title="Card title" description="This is the description"/>
>         </Card>
>     </Col>
>     
>     <Col style={{ display: 'flex' }}>
>         <Card
>             className='flexible-card'
>             style={{ display: 'flex', flexDirection: 'column' }}
>             actions={[
>                 <SettingOutlined key="setting" />,
>                 <EditOutlined key="edit" />,
>                 <EllipsisOutlined key="ellipsis" />,
>             ]}
>         >
>              <Meta title="Card title" description="This is an even longer description to show the card body growing"/>
>         </Card>
>     </Col>
>     
> </Row>
> ```
> 
> And have written in your css :
> 
> ```
> .flexible-card .ant-card-body {
>     flex-grow: 1
> }
> ```
> 
> Instead of this: ![image](https://user-images.githubusercontent.com/3034969/159312990-1965844f-7df1-4846-b30d-181ff570cc68.png)
> 
> You will get this: ![image](https://user-images.githubusercontent.com/3034969/159313817-bbc23ab3-f574-4853-8083-b898efe17da8.png)

Thanks a lot. It helped.
